packages/client/src/components/SafeSettings/components/Owners/AddSafeOwner.js

```diff
@@ -1,90 +1,111 @@
 import React, { useState, useContext } from "react";
+import { useModalContext } from "contexts";
 import { useAddressValidation } from "hooks";
 import { isAddr, formatAddress } from "utils";
 import Svg from "library/Svg";
+import EditThreshold from "../EditThreshold";
 import { Web3Context } from "contexts/Web3";
 
-const AddSafeOwner = ({ onCancel, onNext, safeOwners }) => {
+const AddSafeOwner = ({ treasury, safeOwners }) => {
   const web3 = useContext(Web3Context);
+  const { openModal, closeModal } = useModalContext()
   const { isAddressValid } = useAddressValidation(web3.injectedProvider);
   const [name, setName] = useState("");
   const [address, setAddress] = useState("");
   const [addressValid, setAddressValid] = useState(false);
 
   const onAddressChange = async (newAddress) => {
     setAddress(newAddress);
-    setAddressValid(
-      isAddr(newAddress) &&
-        (await isAddressValid(newAddress)) &&
-        !isAddressExisting(safeOwners, newAddress)
-    );
+    const isValid = isAddr(newAddress) &&
+      (await isAddressValid(newAddress)) &&
+      !isAddressExisting(safeOwners, newAddress)
+    setAddressValid(isValid);
   };
 
   const isAddressExisting = (safeOwners, newAddress) => {
     const address = formatAddress(newAddress);
     return (
-      safeOwners.filter((obj) => obj.address === address && obj.verified)
-        .length !== 0
+      safeOwners.filter((obj) => obj.address === address && obj.verified).length !== 0
     );
   };
 
   const onNextClick = () => {
-    onNext({
-      name,
-      address,
-    });
+    openModal(
+      <EditThreshold treasury={treasury} newOwner={{ address, name }} />
+    )
   };
 
-  const nextButtonClasses = [
-    "button flex-1 is-primary",
-    addressValid ? "" : "disabled",
-  ];
+  const onCancelClick = () => closeModal()
+
+  const addrInputClasses = () => {
+    let classes = [
+      "input p-4 rounded-sm column is-full is-size-6"
+    ]
+    if (addressValid) {
+      classes.push("is-success");
+    }
+    else if (address.length !== 0) {
+      classes.push("is-error")
+    }
+    return classes;
+  }
+
+  const nextButtonClasses = () => {
+    let classes = [
+      "button flex-1 is-primary ",
+      addressValid ? "has-text-weight-bold" : "disabled",
+    ]
+    return classes;
+  }
 
   return (
     <>
       <div className="p-5">
-        <h2 className="is-size-4 has-text-black">Add a new safe owner</h2>
-        <p className="has-text-grey">
-          Heads up, they'll have full access to this safe
-        </p>
+        <h2 className="is-size-4 has-text-black">Add Owner</h2>
       </div>
```

Comment:
This can be removed, when the title is added as a config through openModal, then it will be properly styled and have an X for closing.